src/main/kotlin/com/greenbird/utilihive/stubs/sftp/SftpServerTestContext.kt

```diff
@@ -0,0 +1,323 @@
+        content: String,
+        encoding: Charset
+    ) {
+        val contentAsBytes = content.toByteArray(encoding)
+        putFile(path, contentAsBytes)
+    }
+
+    /**
+     * Puts a file to the SFTP folder. The file is available by the specified path.
+     * @param path the path to the file
+     * @param content the file's content
+     * @throws IOException if the file cannot be written
+     */
+    @Throws(IOException::class)
+    fun putFile(
+        path: String,
+        content: ByteArray
+    ) {
+        verifyWithSftpServerIsNotFinished("upload file")
+        val pathAsObject = fileSystem.getPath(path)
+        ensureDirectoryOfPathExists(pathAsObject)
+        Files.write(pathAsObject, content)
+    }
+
+    /**
+     * Puts a file to the SFTP folder. The file is available by the specified
+     * path. The file's content is read from an `InputStream`.
+     * @param path the path to the file
+     * @param inputStream an `InputStream` that provides the file's content
+     * @throws IOException if the file cannot be written or the input stream cannot be read
+     */
+    @Throws(IOException::class)
+    fun putFile(
+        path: String,
+        inputStream: InputStream
+    ) {
+        verifyWithSftpServerIsNotFinished("upload file")
+        val pathAsObject = fileSystem.getPath(path)
+        ensureDirectoryOfPathExists(pathAsObject)
+        Files.copy(inputStream, pathAsObject)
+    }
+
+    /**
+     * Creates a directory on the SFTP server.
+     * @param path the directory's path
+     * @throws IOException if the directory cannot be created
+     */
+    @Throws(IOException::class)
+    fun createDirectory(
+        path: String
+    ) {
+        verifyWithSftpServerIsNotFinished("create directory")
+        val pathAsObject = fileSystem.getPath(path)
+        Files.createDirectories(pathAsObject)
+    }
+
+    /**
+     * Create multiple directories on the SFTP server.
+     * @param paths the directories' paths.
+     * @throws IOException if at least one directory cannot be created.
+     */
+    @Throws(IOException::class)
+    fun createDirectories(
+        vararg paths: String
+    ) {
+        for (path in paths)
+            createDirectory(path)
+    }
+
+    /**
+     * Gets a text file's content from the SFTP server. The content is decoded
+     * using the specified encoding.
+     * @param path the path to the file
+     * @param encoding the file's encoding
+     * @return the content of the text file
+     * @throws IOException if the file cannot be read
+     */
+    @Throws(IOException::class)
+    fun getFileContent(
+        path: String,
+        encoding: Charset
+    ): String = getFileContent(path).toString(encoding)
+
+    /**
+     * Gets a file from the SFTP server.
+     * @param path the path to the file
+     * @return the content of the file
+     * @throws IOException if the file cannot be read
+     */
+    @Throws(IOException::class)
+    fun getFileContent(
```

Comment:
As mentioned in another comment, overloads (or in this case, almost-overloads) are messy and old school.
Here we see an example why. Having two functions with the same name, but different return types, having their identity being determined based on whether they take an encoding argument or not is not very descriptive API. 

How about `getFileBytes` and `getFileText` instead? 
BTW what happens if the given path points to a directory?
Maybe we should also consider adding a function to return a Path. I guess it can be useful for the users to list directories on the sever etc.?

Reply:
CHanged to getFileBytes and getFileText.
If path points to a directory, getFileBytes or getFileText throws an exception.
Added a function returning Path.